### PR TITLE
Update mypy version from 1.11.2 to 1.13.0

### PR DESCRIPTION
**Pull request by dl-lim**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by dl-lim** [comment]
Man Group London Hackathon 2024

@simonjayhawkins 

@dl-lim 
@ofsouzap
@nvasilev
@shengjie2013
@pyfra
@YasithDSL

**Comment by dl-lim** [comment]
Man Group London Hackathon 2024

@simonjayhawkins 

@ofsouzap
@nvasilev
@shengjie2013
@pyfra
@YasithDSL
@dl-lim

File: pandas/core/dtypes/dtypes.py [hunk 1]
```diff
@@ -2130,9 +2130,11 @@ def _get_common_dtype(self, dtypes: list[DtypeObj]) -> DtypeObj | None:
                 PerformanceWarning,
                 stacklevel=find_stack_level(),
             )
-
         np_dtypes = (x.subtype if isinstance(x, SparseDtype) else x for x in dtypes)
-        return SparseDtype(np_find_common_type(*np_dtypes), fill_value=fill_value)
+        # error: Argument 1 to "np_find_common_type" has incompatible type
```
Comment (simonjayhawkins) [inline pandas/core/dtypes/dtypes.py:2134]: ideally the comment should be immediately above the line where the `#type: ignore` is added

File: pandas/core/nanops.py [hunk 2]
```diff
@@ -738,7 +740,7 @@ def nanmedian(values, *, axis: AxisInt | None = None, skipna: bool = True, mask=
 
     Returns
     -------
-    result : float
+    result : float | ndarray
```
Comment (simonjayhawkins) [inline pandas/core/nanops.py:743]: can you comment in PR a example returning an array

**Review by simonjayhawkins — APPROVED** [review]
Thanks @ofsouzap @nvasilev @shengjie2013 @pyfra @YasithDSL @dl-lim

**Pull request merged**